### ISSUE TITLE
sokol_app.h: Low priority feature request: Rendering waiting on Events

### ISSUE THREAD

**Issue by OetkenPurveyorOfCode** [floooh/sokol#1143]
Since sokol can be used as a backend for some immediate UI frameworks, I was wondering whether there could be a way in sokol_app.h to redraw only when events come in.
Similar to `glfwWaitEvents()` or raylibs
`void EnableEventWaiting(void)`.

The reason to do this is that some UIs are mostly static and it is more efficient to wakt for event instead of polling and rendering every frame.

I haven't seen a way to do this, there is no flag in sapp_desc to call frame only when events come in and otherwise block, and rendering in event_cb and passing NULL for frame_cb is most probably wrong.

If its difficult to implement, don't bother. I can render every frame and it does not matter much. It's an unimportant toy UI anyway.

**Comment by floooh** [comment]
There's sort of a similar issue here:

https://github.com/floooh/sokol/issues/301

...might be difficult to support on all platforms though which have a frame callback mechanism in the window system (e.g. it depends on such a pause/continue feature also being available for that system-provided frame callback).

**Comment by OetkenPurveyorOfCode** [comment]
Might be interesting to see what winit does.
They support a waitevents and pollevents control flow in the event loop and support all platforms.
I agree that #301 is similar though and probably more flexible as you can switch between polling and waiting at runtime.
